components/scream/src/physics/shoc/tests/shoc_run_and_cmp.cpp

```diff
@@ -15,44 +15,77 @@ using namespace scream;
 using namespace scream::util;
 using namespace scream::shoc;
 
+  /* shoc_run_and_cmp can be run in 2 modes. First, generate_baseline 
+   * runs the baseline (aka reference, probably git master) version of 
+   * the code and saves its output as a raw binary file. Then run_and_cmp
+   * runs the new/experimental version of the code and compares it against
+   * the baseline data you've saved to file. Both baseline and cmp modes
+   * start from an initial condition in ../shoc_ic_cases.cpp. Each call to 
+   * shoc_main loops through nadv=15 steps. On top of this, shoc_main is 
+   * called iteratively num_iters=10 steps, performing checks and potentially
+   * writing output each time. This means that shoc_run_and_cmp is really 
+   * a single 150-step shoc run.
+   */
+  
 template <typename Scalar>
 static Int compare (const std::string& label, const Scalar* a,
                     const Scalar* b, const Int& n, const Real& tol) {
-  Int nerr = 0;
+  /* Given a column of data for variable "label" from the reference run 
```

Comment:
It's customary (and required if using, say, Doxygen) to put this sort of description above a function, indented at the same level. But the words are great!